sentry-android-core/src/main/java/io/sentry/android/core/PerformanceAndroidEventProcessor.java

```diff
@@ -0,0 +1,35 @@
+package io.sentry.android.core;
+
+import io.sentry.EventProcessor;
+import io.sentry.protocol.MeasurementValue;
+import io.sentry.protocol.SentryTransaction;
+import java.util.Map;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public final class PerformanceAndroidEventProcessor implements EventProcessor {
+
+  @Override
+  public @Nullable SentryTransaction process(
+      @NotNull SentryTransaction transaction, @Nullable Object hint) {
+    if (!AppStartState.getInstance().isSentStartMetric()) {
+      Long appStartUpInterval = AppStartState.getInstance().getAppStartInterval();
+      if (appStartUpInterval != null) {
+        MeasurementValue value = new MeasurementValue((float) appStartUpInterval);
+
+        String appStartKey =
+            AppStartState.getInstance().getColdStart() ? "app_start_cold" : "app_start_warm";
+
+        transaction.getMeasurements().put(appStartKey, value);
+        AppStartState.getInstance().setSentStartUp();
+      }
+    }
+    final Map<String, @NotNull MeasurementValue> framesMetrics =
+        ActivityFramesState.getInstance().getMetrics();
+    if (framesMetrics != null) {
+      transaction.getMeasurements().putAll(framesMetrics);
```

Comment:
slow/frozen frames are always sent, to every transaction, and represents the overall amount of the app lifecycle, and not from the start-end time of the transaction, we can't track only during the start/end of the transaction unless we implement the whole counting ourselves and this changes based on OS version and so on

Reply:
Whenever we start a transaction, we could check the current count of the frames in `ActivityFramesState`. When you finish a transaction, you again call `ActivityFramesState`, and then you can calculate the different types of frames during the time of the transaction.

Reply:
that would not scale well if I have multiple transactions at the same time, the same issue as counting on iOS the frames on iOS, we can do it next maybe?

Reply:
Running into a similar issue on RN as well. Considered @philipphofmann's suggestion at first but on JS users can change the start/end timestamps which makes it not work and way more difficult to implement metrics at the start/end of the transaction.

Reply:
Why do we run into scaling problems? Hmm, changing it afterward might cause problems. Having it different on platforms is also not a good idea. The advantage of always sending the total number of frames during the lifetime of an application gives you more insights on how many slow frames and frozen frames you have, but the downside is that you have no clue where they are coming from. Being able to correlate them with transactions gives you way more insight. Maybe it would even make sense to send both lifetime frames and frames during the transaction.